### PR TITLE
Allow verify to take multiple keys

### PR DESCRIPTION
**Pull request by quintinm-dev**
key rotation is unsupported: https://github.com/auth0/node-jsonwebtoken/issues/458

this fixes that for `jwt.verify(token, secretOrPublicKey, [options, callback])`
- secretOrPublicKey can be an array of keys
    - keys can be strings or buffers
    - array cannot be used with an unsigned token
    - array must be non-empty
- if options.algorithms is unset:
    - errors if token algorithm unsupported by every key in array
    - otherwise, only attempts to verify against key if algorithm supported by that key

### CODE REVIEW

**Comment by hmcfletch** [comment]
try making `secretOrPublicKey` an array instead of introducing a new options

**Comment by quintinm-dev** [comment]
updated original comment to summarize changes

**Comment by quintinm-dev** [comment]
updated PR @hmcfletch

File: verify.js [hunk 1]
```diff
@@ -97,34 +100,56 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
...
-      return done(new JsonWebTokenError('invalid algorithm'));
+    if (!~supportedAlgorithms.indexOf(decodedToken.header.alg)) {
+      return done(new JsonWebTokenError('invalid algorithm for every given key or algorithm option'));
     }
 
     var valid;
-
     try {
-      valid = jws.verify(jwtString, decodedToken.header.alg, secretOrPublicKey);
+      if (supportedAlgorithms[0] === 'none') {
```
Comment (quintinm-dev) [inline verify.js:146]: comment this

File: verify.js [hunk 2]
```diff
@@ -97,34 +100,56 @@ module.exports = function (jwtString, secretOrPublicKey, options, callback) {
...
     if (!hasSignature && !options.algorithms) {
-      options.algorithms = ['none'];
+      supportedAlgorithms = ['none']
     }
 
-    if (!options.algorithms) {
-      options.algorithms = ~secretOrPublicKey.toString().indexOf('BEGIN CERTIFICATE') ||
-        ~secretOrPublicKey.toString().indexOf('BEGIN PUBLIC KEY') ? PUB_KEY_ALGS :
-        ~secretOrPublicKey.toString().indexOf('BEGIN RSA PUBLIC KEY') ? RSA_KEY_ALGS : HS_ALGS;
+    if (options.algorithms) {
```
Comment (quintinm-dev) [inline verify.js:122]: move this above line 122 to match old control flow

**Comment by quintinm-dev** [comment]
updated @hmcfletch

**Pull request merged**